### PR TITLE
Introduce i_async_batcher and i_payload_transformer interfaces.

### PR DESCRIPTION
**Pull request by kumpera**
The plan is to use i_async_batcher to abstract over the collection serializer used.
We'll use that to have a serializer that applies dedup.

i_payload_transformer will be used to apply compression to each event.

### CODE REVIEW

File: rlclientlib/logger/logger_facade.cc [hunk 1]
```diff
@@ -11,28 +11,37 @@ namespace reinforcement_learning {
       RETURN_ERROR_ARG(nullptr, status, protocol_not_supported, "Current protocol version is not supported");
     }
 
+    template<typename T>
+    i_async_batcher<T>* create_async_batcher(const utility::configuration& c, i_message_sender* sender, utility::watchdog& watchdog, error_callback_fn* perror_cb) {
+      return new async_batcher<T, fb_collection_serializer>(
+        sender,
+        watchdog,
+        perror_cb,
+        c.get_int(name::INTERACTION_SEND_HIGH_WATER_MARK, 198 * 1024),
```
Comment (ataymano) [inline rlclientlib/logger/logger_facade.cc:20]: this factory method does not seem INTERACTION specific.
Do we want to fix https://github.com/VowpalWabbit/reinforcement_learning/issues/172 before?

Comment (kumpera) [reply rlclientlib/logger/logger_facade.cc:20]: uggg, yeah, this won't work as is. I'll take a run at that issue.

**Comment by kumpera** [comment]
@ataymano this PR cleans up the config bits properly: https://github.com/VowpalWabbit/reinforcement_learning/pull/211
I'm going to rebase this PR on top of #211

File: rlclientlib/logger/logger_facade.cc (hunk 1 above)
Comment (kumpera) [reply rlclientlib/logger/logger_facade.cc]: PR rebased on top of master with the batcher config cleanup.

**Comment by kumpera** [comment]
Hmm, I'm probably not going to use this as there might be better abstractions.

**Pull request closed without merging**